src/lib/mgba/websocket-client.ts

```diff
@@ -3,13 +3,47 @@
  * Handles communication with the mGBA Lua HTTP server's WebSocket endpoint
```

Comment:
we need to reduce the complexity of this whole file. it went way too overboard. we need to heavily simplify and refactor for conciseness. for example the whole caching system can be thrown out. all we want are 1. direct reads 2. memory watching 3. shared buffer using memory watching 4. reads that use the shared memory buffer instead of direct reads if that region happens to be watched and a shared buffer.